### PR TITLE
Also consider overlapping variants at different sites.

### PR DESCRIPTION
**Pull request by deflaux**
Continuing to iterate on the variant merge logic.

### CODE REVIEW

**Comment by dionloy** [comment]
LGTM

File: pom.xml [hunk 1]
```diff
@@ -318,9 +308,9 @@
     <junit.version>4.11</junit.version>
     <google.api.version>1.22.0</google.api.version>
     <google.auth.version>0.4.0</google.auth.version>
-    <google.api.genomics.version>v1-rev79-1.22.0</google.api.genomics.version>
+    <google.api.genomics.version>v1-rev87-1.22.0</google.api.genomics.version>
     <google.guava.version>19.0</google.guava.version>
-    <io.grpc.version>0.14.0</io.grpc.version>
+    <io.grpc.version>0.14.1</io.grpc.version>
```
Comment (pgrosu) [inline pom.xml]: Thanks for the bump, so I'll close https://github.com/googlegenomics/utils-java/issues/98 but I'll leave https://github.com/googlegenomics/dataflow-java/issues/205 as `dataflow-java` would need also be updated to pick up the `utils-java` update.

File: src/main/java/com/google/cloud/genomics/utils/grpc/MergeAllVariantsAtSameSite.java [hunk 2]
```diff
@@ -55,78 +57,146 @@ public void merge(Long windowStart, Iterable<Variant> variants, VariantEmitterSt
...
+      if (VariantUtils.isOverlapping(variantRecord, currentStartVariantRecords.get(0))) {
+        priorCallSetNamesBuilder.addAllValues(Lists.transform(variantRecord.getCallsList(),
+            GET_CALL_SET_NAME));
+      } else if (variantRecord.getStart() < currentStartVariantRecords.get(0).getStart()) {
+        // Remove the current element from the iterator and the list since it is
+        // left of this position.
+        iterator.remove();
+      } else {
+        // Break out of this loop since we are now examining records to the right of this position.
+        break;
```
Comment (pgrosu) [inline src/main/java/com/google/cloud/genomics/utils/grpc/MergeAllVariantsAtSameSite.java]: Are we assuming that the variants are always sorted by start position?  Can't some collection of variants come in random order?  If so, breaking early would eliminate some qualified variants.

Comment (deflaux) [inline src/main/java/com/google/cloud/genomics/utils/grpc/MergeAllVariantsAtSameSite.java]: Here's where the sort occurs: https://github.com/deflaux/utils-java/blob/4ada4298122adddc222fd3a79de831ed2b8c0a5c/src/main/java/com/google/cloud/genomics/utils/grpc/MergeAllVariantsAtSameSite.java#L58

Comment (pgrosu) [inline src/main/java/com/google/cloud/genomics/utils/grpc/MergeAllVariantsAtSameSite.java]: Ah, perfect!  I somehow missed that line - thank you for point me to it.

~p

**Pull request merged**